buildSrc/src/main/kotlin/com/datadog/gradle/plugin/gitclone/GitCloneDependenciesTask.kt

```diff
@@ -0,0 +1,159 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.gradle.plugin.gitclone
+
+import com.datadog.gradle.utils.execShell
+import java.io.File
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+open class GitCloneDependenciesTask : DefaultTask() {
+
+    internal var extension: GitCloneDependenciesExtension =
+        GitCloneDependenciesExtension()
+
+    init {
+        group = "datadog"
+        description = "Clones the source code of a dependency into this module"
+        outputs.upToDateWhen { false }
+    }
+
+    // region Task
+
+    @TaskAction
+    fun applyTask() {
+        extension.dependencies.forEach {
+            cloneDependency(it)
+        }
+    }
+
+    private fun cloneDependency(
+        dependency: GitCloneDependenciesExtension.Dependency
+    ) {
+        val target = createTempDir()
+        val copyFrom = if (dependency.originSubFolder.isEmpty()) {
+            target
+        } else {
+            File("${target.absolutePath}${File.separator}${dependency.originSubFolder}")
+        }
```

Comment:
I would put this after `cloneRepository`...is more readable this way.

Reply:
why not 🤷‍♂